confluent_kafka/admin/__init__.py

```diff
@@ -573,3 +573,57 @@ def __repr__(self):
 
     def __str__(self):
         return "{}".format(self.id)
+
+
+class GroupMember(object):
+    """Group member information
+
+    For more information on metadata format, see
+    https://cwiki.apache.org/confluence/display/KAFKA/A+Guide+To+The+Kafka+Protocol#AGuideToTheKafkaProtocol-GroupMembershipAPI
+
+    This class is typically not user instantiated.
+
+    :ivar str id: Member id (generated by broker)
+    :ivar str client_id: Client id
+    :ivar str client_host: Client hostname
+    :ivar bytes metadata: Member metadata(binary), format depends on protocol type
+    :ivar bytes assignment: Member assignment(binary), format depends on protocol type
```

Comment:
Would be great to have APIs to parse `metadata` and `assignment`.

cc https://github.com/edenhill/librdkafka/issues/2643

Reply:
That'd be great, but could be provided as a separate python utility package, perhaps auto-generated from the AK json protocol specs.

Reply:
I'm interested in decoding this data... is this work in progress somewhere? Or maybe point me to the spec you're referring to (I couldn't find it with limited googling) and I can take a crack at it.

Reply:
Found the spec:
https://kafka.apache.org/protocol

Reply:
@snagafritz  What's you solution to decode it? Could you share it out?

Reply:
```python
def parse_member_assignment(buf):
    '''Parse Kafka group assignment'''
    import struct
    assigns = []
    if not buf:
        return assigns
    index = struct.calcsize('>hi')
    _version, assign_len = struct.unpack('>hi', buf[0:index])
    for _ in range(assign_len):
        size = struct.calcsize('>h')
        str_len = struct.unpack('>h', buf[index:index + size])[0]
        index += size
        topic = buf[index: index + str_len].decode('utf-8')
        index += str_len;
        size = struct.calcsize('>i')
        part_len = struct.unpack('>i', buf[index:index + size])[0]
        index += size
        parts = []
        for _ in range(part_len):
            part = struct.unpack('>i', buf[index:index + size])[0]
            parts.append(part)
            index += size
        assigns.append((topic, parts))
    return assigns
```

Reply:
@messense Very helpful! Thanks